File: BFS And DFS ALgorithm/Queue.cpp
#include<bits/stdc++.h>
using namespace std;
class Node
{
public:
    int data;
    Node *next;
    Node(int val)
    {
        data = val;
        next = nullptr;
    }
};
class Queue{
    Node *head;
    public:
    Queue(){
        head=nullptr;
    }
    bool isempty(){
        if(head==nullptr){
            return true;
        }
        return false;
    }
    void enque(int val)
    {
        if (head == NULL)
        {
            head=new Node(val);
            return;
        }
        Node *newNode = new Node(val);
        Node *temp = head;
        while (temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newNode;
    }
    int deque(){
        if(isempty()){
            return -1;
        }
        else{
            int n=head->data;
            head=head->next;
            return n;
        }
    }
    void display(){
        Node *temp = head;
        while (temp != nullptr)
        {
            cout << temp->data;
            if(temp->next!=nullptr){
                cout<<",";
            }
            temp = temp->next;
        }
    }
    
};
int main()
{
    Queue q;
    q.enque(1);
    q.enque(2);
    q.enque(3);
    q.enque(4);
    // q.display();
    cout<<q.deque();
    cout<<q.deque();
    cout<<q.deque();
    cout<<q.deque();
    cout<<q.deque();
    return 0;
}

File: Max Flow/Queue.java
class QNode{
    int v1;
    int v2;
    int weight;
    QNode next;
    public QNode(int v1,int v2,int weight){
        this.v1=v1;
        this.v2=v2;
        this.weight=weight;
        next=null;
    }
}
class Qu{
    QNode head;
    QNode tail;
    public Qu(){
        head=null;
        tail=null;
    }
    boolean isempty()
    {
        if(head==null && tail==null){
            return true;
        }
        return false;
    }
    public void enque(int v1,int v2,int weight){
        QNode temp=new QNode(v1,v2,weight);
        if(head==null){
            this.head=this.tail= temp;
            return;
        }
            System.out.println("Hello");
            this.tail.next= temp;
            this.tail=temp;
            return;
    }
    public int deque(){
        if(isempty()){
            return -1;
        }
        else{
            int n=head.weight;
            head=head.next;
            return n;
        }
    }
    public void display(){
        QNode temp=head;
        while(temp!=null){
            System.out.print("(" + temp.v1 + "," + temp.weight + ")" + " ");
            temp=temp.next;
        }
    }
}
public class Queue {
    public static void main(String[] args) {
        Qu q=new Qu();
        q.enque(0,0,1);
        q.enque(0,0,2);
        q.enque(0,0,3);
        q.enque(0,0,4);
        q.enque(0,0,5);
        q.display();
        System.out.println();
        System.out.print(q.deque());
    }
}
